### PR TITLE
Issue 585 cn lo h is experimental

### LINKED ISSUE #585: Emphasize that CNLoH caller is experimental/unevaluated
- [x] Change docs in the CLI
- [x] Rename tool to CallAllelicSplits
- [x] Change package and class names
- [x] Update FH task configuration or file an issue
- [x] Update Queue script with better naming
- [x] Update wdl

### PR DESCRIPTION
**Pull request by LeeTL1220**
closes #585

### CODE REVIEW

**Comment by LeeTL1220** [comment]
@samuelklee This only shows a failure due to coveralls issue.  Can you review anyway?

**Comment by LeeTL1220** [comment]
@samuelklee I need to merge & rebase commits ... one sec...

**Comment by LeeTL1220** [comment]
@samuelklee done... please review

**Comment by LeeTL1220** [comment]
@samuelklee Just coverage is failing for no good reason....  Can you review?

**Comment by samuelklee** [comment]
I'm not sure about the use of AllelicSplit vs. AllelicBalance throughout this PR.  For example, the output directory and task name in the WDL are call_balanced_dir (which

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 1]
```diff
@@ -230,14 +230,14 @@ workflow case_gatk_acnv_workflow {
           mem=4
     }
 
-    call CNLoHAndSplitsCaller {
+    call AllelicBalanceCaller {
```
Comment (samuelklee) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: The use of "split" vs. "balance" throughout this PR is somewhat confusing.  For example, the output directory and task name in the WDL are call_balanced_dir and AllelicBalanceCaller, respectively, but the CLI is named CallAllelicSplits.

I think it is clear what a "balanced call" should be (is this segment balanced or not?).  It's less clear what a "split" is supposed to represent from the name alone; from AllelicSplitCall.java, it seems to represent quite a lot of quantities.  Perhaps choose a more evocative name?

Note that this was probably a consequence of the fact that "CNLoHCall" was already trying to represent more than its name suggests.

**Comment by samuelklee** [comment]
Back to @LeeTL1220.  If I understand the main intent of this PR, it is simply to add documentation that the CNLoH caller is experimental and to change the name of the tool to reflect that fact.  However, I think the additional refactoring and some of the name choices are a little confusing.

Given that this code is probably going to be deprecated soon, I'll leave it up to you as to how you want to proceed.  I'd actually be fine with simply renaming the tool "ConvertACNVResults" or something along those lines and noting in the documentation that it is primarily meant for internal use, if you are worried that someone will see the original name and try to use it to call CNLoH.

**Comment by LeeTL1220** [comment]
@samuelklee Actually, marketing it as a conversion tool is the best way.  Good idea!  I'll change it.

File: scripts/acnv_cromwell_tests/case_gatk_acnv_workflow_exome.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
   "case_gatk_acnv_workflow.PadTargets.padding": "250",
   "case_gatk_acnv_workflow.disable_sequence_dictionary_validation": "true", 
   "case_gatk_acnv_workflow.plots_dir": "/tmp/plots/",
-  "case_gatk_acnv_workflow.call_cnloh_dir": "/tmp/cnloh/",
+  "case_gatk_acnv_workflow.call_balanced_dir": "/tmp/cnloh/",
```
Comment (LeeTL1220) [inline scripts/acnv_cromwell_tests/case_gatk_acnv_workflow_exome.json]: Done

File: scripts/acnv_cromwell_tests/case_gatk_acnv_workflow_wgs.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
   "case_gatk_acnv_workflow.PadTargets.padding": "250",
   "case_gatk_acnv_workflow.disable_sequence_dictionary_validation": "true",
   "case_gatk_acnv_workflow.plots_dir": "/tmp/plots/",
-  "case_gatk_acnv_workflow.call_cnloh_dir": "/tmp/cnloh/",
+  "case_gatk_acnv_workflow.call_balanced_dir": "/tmp/cnloh/",
```
Comment (LeeTL1220) [inline scripts/acnv_cromwell_tests/case_gatk_acnv_workflow_wgs.json]: Done

Comment (LeeTL1220) [inline scripts/acnv_cromwell_tests/case_gatk_acnv_workflow_wgs.json]: Done

File: scripts/wdl/case_gatk_acnv_workflow.wdl (hunk 1 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: The focus is more about conversion than the calling.

**Comment by LeeTL1220** [comment]
@samuelklee Back at you, though automated tests are still running.  I'll address if failure.

**Comment by LeeTL1220** [comment]
@samuelklee Seems like travis has disappeared, but I added a commit to fix the failing test.

**Comment by LeeTL1220** [comment]
https://travis-ci.org/broadinstitute/gatk-protected/builds/169342850 should show the travis build...

**Comment by LeeTL1220** [comment]
@samuelklee Please disregard that link...

**Comment by LeeTL1220** [comment]
@samuelklee Just a coverage failure again... Can you review?

File: src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java [hunk 4]
```diff
@@ -1,29 +1,29 @@
-package org.broadinstitute.hellbender.tools.exome.cnlohcaller;
+package org.broadinstitute.hellbender.tools.exome.conversion.allelicbalancecaller;
 
 import htsjdk.samtools.util.Locatable;
 import org.broadinstitute.hellbender.tools.exome.ACNVModeledSegment;
 
 import java.io.Serializable;
 
-public class CNLOHCall implements Serializable, Locatable{
+public class AllelicSplitCall implements Serializable, Locatable {
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java]: Again, still not sure about the name AllelicSplitCall.  To me, this sounds like it should just be the opposite of AllelicBalanceCall.  However, it actually represents AllelicBalanceCall, CNLoHCall, and all of the other quantities.  I'll leave it up to you as to whether or not you want to change it.

File: scripts/wdl/case_gatk_acnv_workflow.wdl [hunk 5]
```diff
@@ -230,14 +230,14 @@ workflow case_gatk_acnv_workflow {
...
-    call CNLoHAndSplitsCaller {
+    call ConvertACNVResults {
       input:
           entity_id=row[0],
           gatk_jar=gatk_jar,
           tumor_hets=BayesianHetPulldownPaired.tumor_hets,
           acnv_segments=AllelicCNV.acnv_final_segs,
           tn_file=TumorNormalizeSomaticReadCounts.tn_file,
-          output_dir="${call_cnloh_dir}CallCNLoH/${row[0]}",
+          output_dir="${conversion_dir}CallBalanced/${row[0]}",
```
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow.wdl]: Missed this...

Done

File: src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java (hunk 4 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java]: @samuelklee How about ``AllelicCall``?

Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java]: (Since it has balance and CNLoH)

**Comment by samuelklee** [comment]
Thanks, back to you @LeeTL1220!  Just going to harp on some naming choices one last time---but since we'll probably remove this code soon, don't sweat them too much.  Feel free to squash and merge after!

File: src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java (hunk 4 above)
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java]: ``AllelicCalls`` is what I meant... plural

Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java]: I'm just going to do AllelicCalls.  I don't want to spend any more time on this.

Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/conversion/allelicbalancecaller/AllelicSplitCall.java]: Sounds good to me!

**Pull request merged**